contracts/strategies/VariableRewardsStrategy.sol

```diff
@@ -0,0 +1,311 @@
+        return _amount.mul(withdrawFeeBips).div(_bip());
+    }
+
+    function reinvest() external override onlyEOA {
+        (Reward[] memory rewards, uint256 estimatedTotalReward) = _checkReward();
+        require(estimatedTotalReward >= MIN_TOKENS_TO_REINVEST, "VariableRewardsStrategy::Reinvest amount too low");
+        _reinvest(rewards);
+    }
+
+    function _convertRewardsIntoWAVAX(Reward[] memory _rewards) private returns (uint256) {
+        uint256 avaxAmount = rewardToken.balanceOf(address(this));
+        for (uint256 i = 0; i < _rewards.length; i++) {
+            address reward = _rewards[i].reward;
+            address swapPair = rewardSwapPairs[reward];
+            uint256 amount = _rewards[i].amount;
+            if (amount > 0) {
+                if (reward == address(rewardToken)) {
+                    uint256 balance = address(this).balance;
+                    if (balance > 0) {
+                        WAVAX.deposit{value: balance}();
+                        avaxAmount = avaxAmount.add(amount);
+                    }
+                } else {
+                    if (swapPair > address(0)) {
+                        avaxAmount = avaxAmount.add(
+                            DexLibrary.swap(amount, reward, address(rewardToken), IPair(swapPair))
+                        );
+                    }
+                }
+            }
+        }
+        return avaxAmount;
+    }
+
+    /**
+     * @notice Reinvest rewards from staking contract to deposit tokens
+     * @dev Reverts if the expected amount of tokens are not returned from the staking contract
+     */
+    function _reinvest(Reward[] memory _rewards) private {
+        _getRewards();
+        uint256 amount = _convertRewardsIntoWAVAX(_rewards);
+
+        uint256 devFee = amount.mul(DEV_FEE_BIPS).div(BIPS_DIVISOR);
+        if (devFee > 0) {
+            rewardToken.safeTransfer(devAddr, devFee);
+        }
+
+        uint256 reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(BIPS_DIVISOR);
+        if (reinvestFee > 0) {
+            rewardToken.safeTransfer(msg.sender, reinvestFee);
+        }
+
+        uint256 depositTokenAmount = _convertRewardTokenToDepositToken(amount.sub(devFee).sub(reinvestFee));
+
+        _stakeDepositTokens(depositTokenAmount);
+        emit Reinvest(totalDeposits(), totalSupply);
+    }
+
+    function _stakeDepositTokens(uint256 _amount) private {
+        require(_amount > 0, "VariableRewardsStrategy::Stake amount too low");
+        _depositToStakingContract(_amount);
+    }
+
+    function _checkReward() internal view returns (Reward[] memory, uint256) {
+        Reward[] memory rewards = _pendingRewards();
+        uint256 estimatedTotalReward = rewardToken.balanceOf(address(this));
+        for (uint256 i = 0; i < rewards.length; i++) {
+            address reward = rewards[i].reward;
+            address swapPair = rewardSwapPairs[rewards[i].reward];
+            uint256 balance = IERC20(reward).balanceOf(address(this));
```

Comment:
initialization for `swapPair` and `balance` can be moved into if statement for gas savings